internal/errors/cassandra_test.go

```diff
@@ -20,8 +20,688 @@ package errors
+			keys := []string{
+				"uuid",
+			}
+			return test{
+				name: "returns cassandra key not found error when key is not found",
+				args: args{
+					keys: keys,
+				},
+				want: want{
+					want: Wrapf(NewErrCassandraNotFoundIdentity(), "cassandra key '%s' not found", keys[0]),
+				},
+			}
+		}(),
+		func() test {
+			keys := []string{
+				"uuid_1",
+				"uuid_2",
+			}
+			return test{
+				name: "returns cassandra keys not found error when keys are not found",
+				args: args{
+					keys: keys,
+				},
+				want: want{
+					want: Wrapf(NewErrCassandraNotFoundIdentity(), "cassandra keys '%s' not found", keys),
+				},
+			}
+		}(),
+		func() test {
+			return test{
+				name: "returns nil when key is empty",
+				args: args{
+					keys: nil,
+				},
+				want: want{
+					want: nil,
+				},
+			}
+		}(),
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(tt *testing.T) {
+			if test.beforeFunc != nil {
+				test.beforeFunc(test.args)
+			}
+			if test.afterFunc != nil {
+				defer test.afterFunc(test.args)
+			}
+			if test.checkFunc == nil {
+				test.checkFunc = defaultCheckFunc
+			}
+
+			got := ErrCassandraNotFound(test.args.keys...)
+			if err := test.checkFunc(test.want, got); err != nil {
+				tt.Errorf("error = %v", err)
+			}
+		})
+	}
+}
+
+func TestErrCassandraGetOperationFailed(t *testing.T) {
+	type args struct {
+		key string
+		err error
+	}
+	type want struct {
+		want error
+	}
+	type test struct {
+		name       string
+		args       args
+		want       want
+		checkFunc  func(want, error) error
+		beforeFunc func(args)
+		afterFunc  func(args)
+	}
+	defaultCheckFunc := func(w want, got error) error {
+		if !Is(got, w.want) {
+			return Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", got, w.want)
+		}
+		return nil
+	}
+	tests := []test{
+		func() test {
+			args := args{
+				key: "uuid",
+				err: New("database error"),
+			}
+			return test{
+				name: "returns wrapped fetch key error when key is `uuid` and error is database error",
+				args: args,
+				want: want{
+					Wrapf(args.err, "error failed to fetch key (%s)", args.key),
+				},
+			}
+		}(),
+		func() test {
+			args := args{
+				key: "",
+				err: New("database error"),
+			}
+			return test{
+				name: "returns wrapped fetch key error when key is empty and error is database error",
+				args: args,
+				want: want{
+					Wrapf(args.err, "error failed to fetch key (%s)", args.key),
+				},
+			}
+		}(),
+		func() test {
+			args := args{
+				key: "uuid",
+				err: nil,
+			}
+			return test{
+				name: "returns fetch key error when key is `uuid` and error is nil",
+				args: args,
+				want: want{
+					want: Errorf("error failed to fetch key (%s)", args.key),
+				},
+			}
+		}(),
+		func() test {
+			args := args{
+				key: "",
+				err: nil,
+			}
+			return test{
+				name: "returns fetch key error when key is empty and error is nil",
+				args: args,
+				want: want{
+					want: Errorf("error failed to fetch key (%s)", args.key),
```

Comment:
is it `nil`?
I suggest to write the error without using `Errorf()` or `Wrapf()` for all the test cases, so that we can compare the return value with more accuracy. 
What do you think?

Reply:
thank you for your suggestion.
> I suggest to write the error without using Errorf() or Wrapf() for all the test cases, so that we can compare the return value with more accuracy.

It looks good. I will fix it soon.
thank you